### DISCUSSION TITLE
Istio operator and isctioctl future plans

### DISCUSSION THREAD

**Discussion by ttsokov** [istio/istio#50352] [General]
Hi all!

I have some questions regarding future plans and directions of Istio operator and istioctl:
1. Do you plan to deprecate Istio operator and replace it by istioctl?
2. Do you plan to make istioctl more programmatically reusable by extracting it in separate Golang module with standard interface? Currently it is mainly used by binary. 

Thanks for the information!

**Comment by howardjohn** [comment]
The IstioOperator in-cluster operator is very strongly discouraged and not recommended for any usage in any form. Whether it will be formally deprecated and/or removed is still under discussion, but its plausible.

There are no plans to make istioctl programatically usable, nor do I think there is much path forward for that. For this kind of extensibility I would probably suggest using Helm charts directly, though it depends on what you are tying to do

**Comment by zirain** [comment]
FYI: there's another project https://github.com/istio-ecosystem/sail-operator may be an option.
